krod/modules/research/web_search/search_providers.py

```
@@ -66,7 +73,52 @@ def __init__(self, fallback_provider: Optional[str] = "serpapi"):
+        # Check if Playwright is installed
+        self._check_playwright_installation()
+    
+    def _check_playwright_installation(self):
+        """Check if Playwright is installed and install if missing."""
+        if not CRAWL4AI_AVAILABLE:
+            self.logger.error("Crawl4AI package not installed. Web crawling will be disabled.")
+            return False
+            
+        try:
+            # Try importing playwright to check if it's installed
+            import playwright
```

Ruff (0.11.9)

`playwright` imported but unused; consider using `importlib.util.find_spec` to test for availability

(F401)